src/magicbox/SuggestionsManager.ts

```diff
@@ -471,21 +504,21 @@ export class SuggestionsManager {
     return null;
   }
 
-  private addSelectedStatus(suggestion: HTMLElement): void {
+  private addSelectedStatus(element: HTMLElement): void {
     const selected = this.element.getElementsByClassName(this.options.selectedClass);
     for (let i = 0; i < selected.length; i++) {
       const elem = <HTMLElement>selected.item(i);
       this.removeSelectedStatus(elem);
     }
-    $$(suggestion).addClass(this.options.selectedClass);
-    this.updateSelectedSuggestion(suggestion.innerText);
```

Comment:
`addSelectedStatus` is called by `processKeyboardSelection`, `processKeyboardPreviewSelection` and `processMouseSelection`. However, `processKeyboardSelection` and `processMouseSelection` already call `this.updateSelectedSuggestion`. Calling `this.updateSelectedSuggestion` here caused it to be called on `processKeyboardPreviewSelection`, which called queries unintentionally.